doc/manual/advanced-topics/post-build-hook.xml

```diff
@@ -72,17 +72,12 @@ trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDS
 
 set -eu
 
-# Shift off arguments to the hook until the only remaining arguments
-# are output paths
-while [ "${1:---}" != "--" ]; do
-  shift
-done
-shift # pop off the "--"
-
-echo "Signing paths" "$@"
-nix sign-paths --key-file /etc/nix/key.secret "$@"
-echo "Uploading paths" "$@"
-exec nix copy --to 's3://example-nix-cache' "$@"
+ALL_OUT_PATHS=$(echo $OUT_PATHS | tr ':' ' ')
```

Comment:
Please use a space (or maybe a newline) as separator in `$OUT_PATHS`. That way you don't need to do any special processing in shell scripts.

Reply:
I'd rather implement [@grahamc's proposal](https://github.com/NixOS/nix/pull/2995#discussion_r305346219), is that OK for you?

Reply:
In the face of bikeshed, listen to @edolstra